circus-train-integration-tests/src/test/java/com/hotels/bdp/circustrain/integration/CircusTrainHdfsHdfsIntegrationTest.java

```diff
@@ -850,6 +851,123 @@ public void checkAssertion() throws Exception {
+  }
+
+  @Test
+  public void unpartitionedTableFullOverwrite() throws Exception {
+    helper.createManagedUnpartitionedTable(toUri(sourceWarehouseUri, DATABASE, SOURCE_MANAGED_UNPARTITIONED_TABLE));
+    LOG.info(">>>> Table {} ", sourceCatalog.client().getTable(DATABASE, SOURCE_MANAGED_UNPARTITIONED_TABLE));
+    // adjusting the sourceTable, mimicking the change we want to update
+    Table sourceTable = sourceCatalog.client().getTable(DATABASE, SOURCE_MANAGED_UNPARTITIONED_TABLE);
+    sourceTable.putToParameters("paramToUpdate", "updated");
+    sourceCatalog.client().alter_table(sourceTable.getDbName(), sourceTable.getTableName(), sourceTable);
+
+    // creating replicaTable
+    final URI replicaLocation = toUri(replicaWarehouseUri, DATABASE, SOURCE_MANAGED_UNPARTITIONED_TABLE);
+    TestUtils
+        .createUnpartitionedTable(replicaCatalog.client(), DATABASE, TARGET_UNPARTITIONED_MANAGED_TABLE,
+            replicaLocation);
+    Table replicaTable = replicaCatalog.client().getTable(DATABASE, TARGET_UNPARTITIONED_MANAGED_TABLE);
+    // setting up parameters and additional columns
+    setupReplicaTable(replicaTable, false, replicaLocation);
+    replicaCatalog.client().alter_table(replicaTable.getDbName(), replicaTable.getTableName(), replicaTable);
+
+    exit.expectSystemExitWithStatus(0);
+    File config = dataFolder.getFile("unpartitioned-single-table-full-overwrite.yml");
+    CircusTrainRunner runner = CircusTrainRunner
+        .builder(DATABASE, sourceWarehouseUri, replicaWarehouseUri, housekeepingDbLocation)
+        .sourceMetaStore(sourceCatalog.getThriftConnectionUri(), sourceCatalog.connectionURL(),
+            sourceCatalog.driverClassName())
+        .replicaMetaStore(replicaCatalog.getThriftConnectionUri())
+        .build();
+    exit.checkAssertionAfterwards(new Assertion() {
+      @Override
+      public void checkAssertion() throws Exception {
+        Table hiveTable = replicaCatalog.client().getTable(DATABASE, TARGET_UNPARTITIONED_MANAGED_TABLE);
+        assertThat(hiveTable.getDbName(), is(DATABASE));
+        assertThat(hiveTable.getTableName(), is(TARGET_UNPARTITIONED_MANAGED_TABLE));
+        assertThat(hiveTable.getParameters().get(REPLICATION_EVENT.parameterName()), startsWith("ctt-"));
+        assertThat(hiveTable.getParameters().get(REPLICATION_MODE.parameterName()), is("FULL_OVERWRITE"));
+        assertThat(hiveTable.getParameters().get("paramToUpdate"), is("updated"));
+        assertThat(isExternalTable(hiveTable), is(true));
+        assertThat(hiveTable.getSd().getCols(), is(DATA_COLUMNS));
+      }
+    });
+    runner.run(config.getAbsolutePath());
+  }
+
+  private void setupReplicaTable(Table replicaTable, boolean partitioned, URI replicaLocation) throws Exception {
+    setupRelicaParameters(replicaTable);
+    if (partitioned) {
+      addPartitionsToReplica(replicaTable, replicaLocation);
+    }
+  }
+
+  private void setupRelicaParameters(Table replicaTable) {
```

Comment:
This function is not just setting params but also adding two new columns to the table. Splitting this function up would be better.